### PR TITLE
fix recursive curve token retrieval

### PR DESCRIPTION
**Pull request by mushroomsforest**
If `coin0` of a curve token like [`crv3crypto`](https://polygonscan.com/token/0xdad97f7713ae9437fa9249920ec8507e5fbb23d3) is another curve token like [`3pool`](https://polygonscan.com/address/0xE7a24EF0C5e95Ffb0f6684b813A78F2a3AD7D171), then current impl will throw `undefined` error during multicall

### CODE REVIEW

File: src/static/js/ethers_helper.js [hunk 1]
```diff
@@ -1190,11 +1190,13 @@ async function getJar(app, jar, address, stakingAddress) {
...
 async function getCurveToken(app, curve, address, stakingAddress, minterAddress) {
   const minter = new ethcall.Contract(minterAddress, MINTER_ABI)
-  const [virtualPrice, coin0] = await app.ethcallProvider.all([minter.get_virtual_price(), minter.coins(0)]);
-  const token = await getToken(app, coin0, address);
-  const calls = [curve.decimals(), curve.balanceOf(stakingAddress), curve.balanceOf(app.YOUR_ADDRESS),
-    curve.name(), curve.symbol(), curve.totalSupply()];
-  const [decimals, staked, unstaked, name, symbol, totalSupply] = await app.ethcallProvider.all(calls);
+  const [virtualPrice, coin0] = await app.ethcallProvider.all([minter.get_virtual_price(), minter.coins(0)]);  
+  
+  const lpToken = new ethcall.Contract(curve.address, CURVE_ABI);
```
Comment (vfat0) [inline src/static/js/ethers_helper.js:1195]: The `curve` var here is the same thing as `lpToken`, not sure what the difference in these changes is

File: src/static/js/ethers_helper.js [hunk 2]
```diff
@@ -1205,7 +1207,7 @@ async function getCurveToken(app, curve, address, stakingAddress, minterAddress)
       unstaked: unstaked  / 10 ** decimals,
       contract: curve,
       tokens : [address, coin0],
-      token,
+      token0,
```
Comment (vfat0) [inline src/static/js/ethers_helper.js:1210]: This might (or might not) break the price calculations so needs to be tested in some curve projects

**Comment by vfat0** [comment]
yeah we confirmed this breaks stakedao and badger on eth

**Comment by mushroomsforest** [comment]
guess we will have a specialized loader for this recursive curve token

**Pull request closed without merging**